src/vmm/src/devices/bus.rs

```diff
@@ -102,97 +102,65 @@ impl ConstantDevice {
 }
 
 impl BusDevice {
-    pub fn i8042_device_ref(&self) -> Option<&I8042Device> {
-        match self {
-            Self::I8042Device(x) => Some(x),
-            _ => None,
-        }
-    }
-    #[cfg(target_arch = "aarch64")]
-    pub fn rtc_device_ref(&self) -> Option<&RTCDevice> {
-        match self {
-            Self::RTCDevice(x) => Some(x),
-            _ => None,
-        }
-    }
-    pub fn boot_timer_ref(&self) -> Option<&BootTimer> {
-        match self {
-            Self::BootTimer(x) => Some(x),
-            _ => None,
-        }
-    }
-    pub fn mmio_transport_ref(&self) -> Option<&MmioTransport> {
-        match self {
-            Self::MmioTransport(x) => Some(x),
-            _ => None,
-        }
-    }
-    pub fn serial_ref(&self) -> Option<&SerialDevice<std::io::Stdin>> {
-        match self {
-            Self::Serial(x) => Some(x),
-            _ => None,
-        }
-    }
-
-    pub fn i8042_device_mut(&mut self) -> Option<&mut I8042Device> {
+    pub fn i8042_device_ref(&self) -> Option<&Arc<Mutex<I8042Device>>> {
         match self {
             Self::I8042Device(x) => Some(x),
             _ => None,
         }
     }
     #[cfg(target_arch = "aarch64")]
-    pub fn rtc_device_mut(&mut self) -> Option<&mut RTCDevice> {
+    pub fn rtc_device_ref(&self) -> Option<&Arc<Mutex<RTCDevice>>> {
         match self {
             Self::RTCDevice(x) => Some(x),
             _ => None,
         }
     }
-    pub fn boot_timer_mut(&mut self) -> Option<&mut BootTimer> {
+    pub fn boot_timer_ref(&self) -> Option<&Arc<Mutex<BootTimer>>> {
         match self {
             Self::BootTimer(x) => Some(x),
             _ => None,
         }
     }
-    pub fn mmio_transport_mut(&mut self) -> Option<&mut MmioTransport> {
+    pub fn mmio_transport_ref(&self) -> Option<&Arc<Mutex<MmioTransport>>> {
         match self {
             Self::MmioTransport(x) => Some(x),
             _ => None,
         }
     }
-    pub fn serial_mut(&mut self) -> Option<&mut SerialDevice<std::io::Stdin>> {
+    pub fn serial_ref(&self) -> Option<&Arc<Mutex<SerialDevice<std::io::Stdin>>>> {
```

Comment:
These could have signatures along the lines of 
```suggestion
    pub fn serial_ref(&self) -> Option<MutexGuard<SerialDevice<std::io::Stdin>>> {
```
by moving the `.lock().expect("Poisoned mutex")` that all call-sites do into the these functions.

Reply:
All done. Thanks for the suggestion, I missed that somehow.